File: docs/snippets/ruby/cache/disk_cache.rb
```ruby
require 'kreuzberg'
require 'fileutils'

cache_dir = File.expand_path('~/.cache/kreuzberg')
FileUtils.mkdir_p(cache_dir)

config = Kreuzberg::ExtractionConfig.new(
  use_cache: true,
  cache_config: Kreuzberg::CacheConfig.new(
    cache_path: cache_dir,
    max_cache_size: 500 * 1024 * 1024,
    cache_ttl_seconds: 7 * 86400,
    enable_compression: true,
  )
)

kreuzberg = Kreuzberg::Client.new(config)

puts "First extraction (will be cached)..."
result1 = kreuzberg.extract_file('document.pdf')
puts "  - Content length: #{result1.content.length}"
puts "  - Cached: #{result1.metadata['was_cached']}"

puts "\nSecond extraction (from cache)..."
result2 = kreuzberg.extract_file('document.pdf')
puts "  - Content length: #{result2.content.length}"
puts "  - Cached: #{result2.metadata['was_cached']}"

puts "\nResults are identical: #{result1.content == result2.content}"

cache_stats = kreuzberg.get_cache_stats
puts "\nCache Statistics:"
puts "  - Total entries: #{cache_stats['total_entries']}"
puts "  - Cache size: #{(cache_stats['cache_size_bytes'] / 1024.0 / 1024.0).round(1)} MB"
puts "  - Hit rate: #{(cache_stats['hit_rate'] * 100).round(1)}%"
```

File: docs/snippets/java/cache/DiskCache.java
```java
import com.kreuzberg.*;
import java.nio.file.Files;
import java.nio.file.Paths;

public final class DiskCache {
  private DiskCache() {}

  public static void main(String[] args) throws Exception {
    String cacheDir = System.getProperty("user.home") + "/.cache/kreuzberg";
    Files.createDirectories(Paths.get(cacheDir));

    CacheConfig cacheConfig = new CacheConfig(
      cacheDir,
      500L * 1024 * 1024,
      7L * 86400,
      true
    );

    ExtractionConfig config = new ExtractionConfig.Builder()
      .useCache(true)
      .cacheConfig(cacheConfig)
      .build();

    Kreuzberg kreuzberg = new Kreuzberg(config);

    System.out.println("First extraction (will be cached)...");
    ExtractionResult result1 = kreuzberg.extractFile("document.pdf");
    System.out.println("  - Content length: " + result1.content().length());
    System.out.println("  - Cached: " + result1.metadata().wasCached());

    System.out.println("\nSecond extraction (from cache)...");
    ExtractionResult result2 = kreuzberg.extractFile("document.pdf");
    System.out.println("  - Content length: " + result2.content().length());
    System.out.println("  - Cached: " + result2.metadata().wasCached());

    System.out.println("\nResults are identical: " + result1.content().equals(result2.content()));

    CacheStats cacheStats = kreuzberg.getCacheStats();
    System.out.println("\nCache Statistics:");
    System.out.println("  - Total entries: " + cacheStats.totalEntries());
    System.out.println("  - Cache size: " + String.format("%.1f", cacheStats.cacheSizeBytes() / 1024.0 / 1024.0) + " MB");
    System.out.println("  - Hit rate: " + String.format("%.1f", cacheStats.hitRate() * 100) + "%");
  }
}
```
